PurchasesTests/Networking/HTTPClientTests.swift

```diff
@@ -544,9 +560,9 @@ class HTTPClientTests: XCTestCase {
         stub(condition: isPath("/v1" + path)) { request in
             usleep(30)
             let requestData = request.ohhttpStubs_httpBody!
-            let requestBodyDict = try! JSONSerialization.jsonObject(with: requestData, options: []) as! [String: Any]
+            let requestBodyDict = try? JSONSerialization.jsonObject(with: requestData, options: []) as? [String: Any]
 
-            let requestNumber = requestBodyDict["requestNumber"] as! Int
+            let requestNumber = requestBodyDict?["requestNumber"] as? Int
```

Comment:
Following the code here is hard to tell if this changes semantics of the test. Maybe `XCTUnwrap` this and the `JSONSerialization`?

Reply:
Yeah, that worried me because I wasn't sure if this would change the semantics of the tests but the problem is that here we can't use `XTUnwrap` because this sentence belongs to a [closure that is not defined as a throwing function](https://github.com/AliSoftware/OHHTTPStubs/blob/12f19662426d0434d6c330c6974d53e2eb10ecd9/Sources/OHHTTPStubsSwift/OHHTTPStubsSwift.swift#L107) 😕

So, I've created a private function to extract the request number using `XCTUnwrap`. You can check it [here](https://github.com/RevenueCat/purchases-ios/blob/7108e5ce0c01cee1b792293baf99d0539074ebc4/PurchasesTests/Networking/HTTPClientTests.swift#L818). So, the semantics are the same and we don't force any casting.

WDYT? 🤔